cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectConfigurationBuilder.java

```diff
@@ -283,20 +287,75 @@ private void printConfigProviders(AbstractConfiguration userConfig) {
+        if (configurations instanceof KafkaConnectConfiguration kcConfiguration) {
+            configurations = new KafkaConnectConfiguration(kcConfiguration);
+        } else if (configurations instanceof KafkaMirrorMaker2Configuration kmm2Configuration) {
+            configurations = new KafkaMirrorMaker2Configuration(kmm2Configuration);
+        } else {
+            // empty user configuration
+            configurations = new KafkaConnectConfiguration(reconciliation, new ArrayList<>());
+        }
+
         printConfigProviders(configurations);
 
-        if (configurations != null && !configurations.getConfiguration().isEmpty()) {
-            printSectionHeader("Provided configurations");
+        printMetricReportersForKey("metric.reporters", configurations, injectKafkaJmxReporter, injectStrimziMetricsReporter);
+        printMetricReportersForKey("admin.metric.reporters", configurations, injectKafkaJmxReporter, injectStrimziMetricsReporter);
+        printMetricReportersForKey("producer.metric.reporters", configurations, injectKafkaJmxReporter, injectStrimziMetricsReporter);
+        printMetricReportersForKey("consumer.metric.reporters", configurations, injectKafkaJmxReporter, injectStrimziMetricsReporter);
+
+        if (!configurations.getConfiguration().isEmpty()) {
+            printSectionHeader("User provided configuration");
             writer.println(configurations.getConfiguration());
             writer.println();
         }
         return this;
     }
 
+    private void printMetricReportersForKey(String configKey,
+                                            AbstractConfiguration userConfig,
+                                            boolean injectKafkaJmxReporter,
+                                            boolean injectStrimziMetricsReporter) {
+        // Build a list of reporters to inject based on flags
+        List<String> reportersToInject = new ArrayList<>();
+        // Since Kafka 4 the JmxReporter is explicitly added when JmxPrometheusExporter is enabled
+        if (injectKafkaJmxReporter) reportersToInject.add("org.apache.kafka.common.metrics.JmxReporter");
+        if (injectStrimziMetricsReporter) reportersToInject.add(StrimziMetricsReporterConfig.KAFKA_CLASS);
+
+        if (!reportersToInject.isEmpty()) {
+            if (userConfig != null && !userConfig.getConfiguration().isEmpty() &&
+                    userConfig.getConfigOption(configKey) != null) {
+                // handle user configuration if present and avoids duplicates
+                String configValue = userConfig.getConfigOption(configKey);
+
+                reportersToInject = reportersToInject.stream()
+                        .filter(r -> !userConfig.getConfigOption(configKey).contains(r))
+                        .toList();
+
+                if (!reportersToInject.isEmpty()) {
+                    userConfig.removeConfigOption(configKey);
+
+                    printSectionHeader(configKey + " configuration");
```

Comment:
I won't repeat myself anymore but I think that as section header we should have a more meaningful title instead of just printing the broker configuration key. We are using this approach across all configuration.